### PR TITLE
Feat/covid app

### PR DESCRIPTION
**Pull request by GlebPilipenko**

### CODE REVIEW

File: src/utils/getFormattedDate.ts [hunk 1]
```diff
@@ -1,14 +1,14 @@
 const oneThousand = 1000;
 
 export const getFormattedDate = (dt: string | number) => {
-  const date = new Date(+dt * oneThousand);
-  const mm = date.getMonth() + 1;
-  const yyyy = date.getFullYear();
-  const dd = date.getDate();
+    const date = new Date(+dt * oneThousand);
```
Comment (aashihmin) [inline src/utils/getFormattedDate.ts]: ?

File: src/widgets/covid/CovidDiagram.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import {FC, useEffect, useState} from 'react';
+import {PropsType} from './typings';
+import {getInfoByCovid} from 'src/api';
+import {ErrorMessage} from 'src/common/errorMessage';
+import {SVGDiagrams} from './components/svgDiagrams';
+import {CovidAPIType} from 'src/api/covidApi/typings';
+
+export const CovidDiagram: FC<PropsType> = ({country = 'invalid_country'}) => {
```
Comment (aashihmin) [inline src/widgets/covid/CovidDiagram.tsx]: strange default value

File: src/widgets/covid/CovidDiagram.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+          : setError('Your request is blocked...');
+      }
+    })();
+  }, [country]);
+
+  if (!state) {
+    return <ErrorMessage errorMessage={error} />;
+  }
+
+  if (state.length > 200) {
```
Comment (aashihmin) [inline src/widgets/covid/CovidDiagram.tsx]: What does it mean?

File: src/widgets/covid/enum/index.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+export enum ArrayLength {
```
Comment (aashihmin) [inline src/widgets/covid/enum/index.ts]: Delete this file

File: src/widgets/covid/components/svgDiagrams/SVGDiagrams.tsx [hunk 5]
```diff
@@ -0,0 +1,89 @@
+import {FC} from 'react';
+import {AxisXY} from '../axisXY';
+import {PropsType} from './typings';
+import style from './SVGDiagrams.module.css';
+import {getCapitalizedString} from 'src/utils';
+import {ColumnWithInformation} from '../columnWithInformation';
+
+export const SVGDiagrams: FC<PropsType> = ({state}) => {
+  const [, , , cases, , deaths, , recovered, , active] = state;
```
Comment (aashihmin) [inline src/widgets/covid/components/svgDiagrams/SVGDiagrams.tsx]: ???

File: src/widgets/covid/CovidDiagram.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+import {DefaultQueryParameters} from 'src/enums';
+import {ErrorMessage} from 'src/common/errorMessage';
+import {SVGDiagrams} from './components/svgDiagrams';
+import {CovidAPIType} from 'src/api/covidApi/typings';
+
+export const CovidDiagram: FC<PropsType> = ({
+  country = DefaultQueryParameters.InvalidCountry
+}) => {
+  const [error, setError] = useState<string>('');
+  const [state, setState] = useState<any | CovidAPIType[]>(null);
```
Comment (aashihmin) [inline src/widgets/covid/CovidDiagram.tsx]: Why do we need any type?

File: src/widgets/covid/CovidDiagram.tsx [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+  const numCountriesInWorld = 193;
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const infoByCovid = await getInfoByCovid(country);
+        setState(Object.entries(infoByCovid.data));
```
Comment (aashihmin) [inline src/widgets/covid/CovidDiagram.tsx]: ???
```suggestion
        if (infoByCovid?.data) {
          const { cases, deaths, recovered, active } = infoByCovid.data;
          setState({ cases, deaths, recovered, active });
        }
```
or smth like this

File: src/widgets/covid/CovidDiagram.tsx [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+          : setError(`Your request is blocked...`);
+      }
+    })();
+  }, [country]);
+
+  if (!state) {
+    return <ErrorMessage errorMessage={error} />;
+  }
+
+  if (state.length >= numCountriesInWorld) {
```
Comment (aashihmin) [inline src/widgets/covid/CovidDiagram.tsx]: strange check..

File: src/widgets/covid/components/svgDiagrams/typings/index.ts [hunk 9]
```diff
@@ -0,0 +1,5 @@
+import {CovidAPIType} from 'src/api/covidApi/typings';
+
+export type PropsType = {
+  state: any | CovidAPIType;
```
Comment (aashihmin) [inline src/widgets/covid/components/svgDiagrams/typings/index.ts]: type any

**Review by aashihmin — CHANGES_REQUESTED** [review]

**Review by aashihmin — APPROVED** [review]

**Review by alexwerst — APPROVED** [review]

**Pull request merged**